src/Microsoft.VisualStudio.Web.CodeGeneration.EntityFrameworkCore/NavigationMetadata.cs

```diff
@@ -32,6 +31,20 @@ public NavigationMetadata(INavigation navigation, Type dbContextType)
                 .Where(p => p.DeclaringEntityType == navigation.DeclaringEntityType)
                 .Select(p => p.Name)
                 .ToArray();
+
+            // The default for the display property is the primary key of the navigation. 
+            DisplayPropertyName = PrimaryKeyNames[0];
+
+            // If there is a non nullable string property in the navigation's target type, we use that instead. 
+            var displayPropertyCandidate = navigation
+                .GetTargetType()
+                .GetDeclaredProperties()
+                .FirstOrDefault(p => !p.IsColumnNullable() && p.ClrType == typeof(string));
```

Comment:
As the name implies `IsColumnNullable()` returns the corresponding column nullability, wouldn't calling `IsNullable()` be more appropriate here?

Also `GetDeclaredProperties()` does not return any inherited properties, shouldn't they be considered as well?